src/main/java/org/walkmod/Options.java

```diff
@@ -144,52 +131,38 @@ public String getConfigurationFormat() {
 		return value.toString();
 	}
 
-	public void setOffline(boolean offline) {
-		this.options.put(OFFLINE, Boolean.valueOf(offline));
-	}
-
 	public boolean isOffline() {
 		Object value = this.options.get(OFFLINE);
 		return value != null && (Boolean) value;
 
 	}
-	
+
 	public String getPath(){
 	    Object path = this.options.get(CHAIN_PATH);
 	    if(path != null){
 	        return path.toString();
 	    }
 	    return null;
 	}
-	
-	public void setPath(String path){
-	    this.options.put(CHAIN_PATH, path);
-	}
-
-	public void setVerbose(boolean verbose) {
-		this.options.put(VERBOSE, Boolean.valueOf(verbose));
-	}
 
 	public boolean isVerbose() {
 		Object value = this.options.get(VERBOSE);
 		return value != null && (Boolean) value;
 	}
 
-	public void setPrintErrors(boolean printErrors) {
-		this.options.put(PRINT_ERRORS, Boolean.valueOf(printErrors));
+	/**
+	 * @return immutable map or null
+	 */
+	public Map<String, Object> getDynamicArgs(){
+		return  (Map<String, Object>) options.get(DYNAMIC_ARGS);
 	}
 
-	public void setExecutionDirectory(File executionDirectory) {
-		options.put(EXECUTION_DIRECTORY, executionDirectory);
-	}
-	
-	public void setDynamicArgs(Map<String, Object> dynamicArgs){
-	    options.put(DYNAMIC_ARGS, dynamicArgs);
-	}
-	
-	public Map<String, Object> getDynamicArgs(){
-	    Object value = options.get(DYNAMIC_ARGS);
-	    return  (Map<String, Object>) value;
+	/**
+	 * @return mutable map or null
+	 */
+	public Map<String, Object> getMutableCopyOfDynamicArgs() {
```

Comment:
Hmm ... 
Options make a statement: I am immutable and all the stuff you get from me is immutable if not
explicitly said otherwise. And the "if" is something I don't like. 
I only added it because of the "null" return of getDynamicArgs.
I would prefer getDynamoicArgs to always return an immutable map, never null and
make a mutable copy of that map at the used places.